service/src/main/java/com/epam/esm/logic/impl/TagServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package com.epam.esm.logic.impl;
+
+import com.epam.esm.dao.TagDao;
+import com.epam.esm.entity.Tag;
+import com.epam.esm.exception.NotFoundEntityException;
+import com.epam.esm.logic.TagService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class TagServiceImpl implements TagService {
+    private final TagDao tagDao;
+
+    @Autowired
+    public TagServiceImpl(TagDao tagDao) {
+        this.tagDao = tagDao;
+    }
+
+    @Override
+    public long create(Tag tag) {
+        return 0;
+    }
+
+    @Override
+    public List<Tag> findAll() {
+        return tagDao.findAll();
+    }
+
+    @Override
+    @Transactional
+    public Tag findById(long id) {
+        Optional<Tag> optionalTag = tagDao.findById(id);
+        if (!optionalTag.isPresent()) {
```

Comment:
Don't start checking a condition with negation. Move to a separate private method.